backend/cfg/cfg.ml

```diff
@@ -191,17 +190,39 @@ let add_block_exn t block =
       Label.format block.start;
   Label.Tbl.add t.blocks block.start block
 
-let remove_block_exn t label =
-  match Label.Tbl.find t.blocks label with
-  | exception Not_found ->
-    Misc.fatal_errorf "Cfg.remove_block_exn: block %a not found" Label.format
-      label
-  | _ -> Label.Tbl.remove t.blocks label
+let remove_trap_instructions t removed_trap_handlers =
+  let remove_trap_instr _ b =
+    DLL.iter_cell b.body ~f:(fun cell ->
+        let i = DLL.value cell in
+        match i.desc with
+        | Pushtrap { lbl_handler } | Poptrap { lbl_handler } ->
+          if Label.Set.mem lbl_handler removed_trap_handlers
+          then DLL.delete_curr cell
+        | Op _ | Reloadretaddr | Prologue | Stack_check _ -> ())
+  in
+  if not (Label.Set.is_empty removed_trap_handlers) then assert Config.flambda2;
+  (* remove Lpushtrap and Lpoptrap instructions that refer to dead labels. *)
+  Label.Tbl.iter remove_trap_instr t.blocks
```

Comment:
We should avoid the iteration when the set empty.

Reply:
fixed.